File: zchan_player_plus/src/main/java/com/example/zchan_player_plus/ui/PlayerActivity.java
package com.example.zchan_player_plus.ui;


import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.example.zchan_player_plus.JniImp;
import com.example.zchan_player_plus.R;
import com.example.zchan_player_plus.view.PlayerSurface;


public class PlayerActivity extends AppCompatActivity {
    private Thread mThread = new Thread(new Runnable() {
        @Override
        public void run() {
            JniImp.startPlay("/sdcard/input.mp4");
        }
    });

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_player_plus);
        ((PlayerSurface)findViewById(R.id.playerSurface)).setOnSurfaceListener(holder -> {
            JniImp.setSurface(holder.getSurface());
            mThread.start();
        });
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        JniImp.stopPlay();
        mThread.interrupt();
    }
}

File: playerCore/src/main/java/com/example/playercore/PlayerActivity.kt
package com.example.playercore

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class PlayerActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_player)
        val playerSurface = findViewById<PlayerSurface>(R.id.playerSurface)
        playerSurface.setOnSurfaceCreated {
            PlayInterface.setSurface(it.surface)
            Thread {
                PlayInterface.startPlay("/data/data/com.example.bilibili/test.mp4")
            }.start()
        }
    }


}